compiler/dyno/lib/types/TupleType.cpp

```diff
@@ -115,7 +123,49 @@ TupleType::getGenericTupleType(Context* context) {
   return getTupleType(context, id, name, instantiatedFrom, subs).get();
 }
 
+const TupleType*
+TupleType::getVarArgTuple(Context* context,
+                          std::vector<QualifiedType> eltTypes) {
+  SubstitutionsMap subs;
+  int i = 0;
+  for (auto t : eltTypes) {
+    subs.emplace(idForTupElt(i), t);
+    i++;
+  }
+
+  auto name = UniqueString::get(context, "_tuple");
+  auto id = ID();
+  const TupleType* instantiatedFrom = getGenericTupleType(context);
+  return getTupleType(context, id, name, instantiatedFrom,
+                      std::move(subs), true).get();
+}
+
+const TupleType*
+TupleType::getVarArgTuple(Context* context,
+                          QualifiedType paramSize,
+                          QualifiedType varArgEltType) {
+  assert(varArgEltType != QualifiedType());
+
+  if (paramSize != QualifiedType() &&
```

Comment:
I'm reviewing commit-by-commit as you recommended, so I'm not sure if this is how the code looks in the final product, but couldn't this be `!paramSize.isUnknown()`? Technically slightly different than this line (behaves differently for `paramSize` that is marked `param` but has no `typePtr`, or that has an `UnknownType` as type), but would avoid constructing a new `QualifiedType`.

Reply:
I think from a readability standpoint that's an improvement. I'll make the change shortly.

I wouldn't expect there to be a noticeable cost for constructing the default QualifiedType, especially given how infrequently we'll build VarArgTuples, but it can't hurt to use the method instead.

Reply:
Also, I don't think a QualifiedType with a ``param`` pointer, but no ``typePtr`` or  an Unknown kind, would be valid. It's certainly possible to create one, but it wouldn't have a valid meaning.